pkg/controller/controller_utils.go

```diff
@@ -418,7 +418,7 @@ type RealRSControl struct {
 var _ RSControlInterface = &RealRSControl{}
 
 func (r RealRSControl) PatchReplicaSet(namespace, name string, data []byte) error {
-	_, err := r.KubeClient.ExtensionsV1beta1().ReplicaSets(namespace).Patch(name, types.StrategicMergePatchType, data)
+	_, err := r.KubeClient.AppsV1().ReplicaSets(namespace).Patch(name, types.StrategicMergePatchType, data)
```

Comment:
I don't understand why this need to be changed. The [test failure](https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/70886/pull-kubernetes-bazel-test/67258/) might just be a flake. Could you discard the change and retest?

Reply:
It is not a flake unfortunately, I was able to reproduce it locally. The reason the test fails is because it uses this helper, which uses the old ExtensionsV1Beta1 interface to RS, which *I think* causes the fake client to be unable to find the AppsV1 RS. The result is that when the deployment tries to patch the RS to adopt it, it can’t find it so it ignores it. 

The reason it worked before is because if the action wasn’t handled, it would return an empty object with a nil error, so it would ‘succeed’

Reply:
mmmmmm.. i guess it should have been done in #70370 @liggitt

Reply:
fake clients don't understand that apps and extensions replicasets are stored in the same place, nor would I expect them to. https://github.com/kubernetes/kubernetes/pull/70370 consistently moves all callers to apps/v1, let's defer this change to that PR

Reply:
Thanks for the explanation. IIUC the `fixture.go` change in this PR is legit, but we are bitten by our debt of having two API groups share the storage, and the code migration being half-way. 

@bouk We can rebase after #70370 is merged, then apply the labels to merge this PR.

Reply:
Sounds good!

Reply:
It won't cause conflict in #70370, as the [change](https://github.com/kubernetes/kubernetes/pull/70370/files#diff-44bba2c68a6ea2b4ca0dbb7f4465828e) there is identical, so I won't block this PR on #70370.

Reply:
verified that all callers of this method are doing things that apply equally well to apps/v1 and extensions/v1beta1 replicasets